### PR TITLE
V2 recovery

### PR DESCRIPTION
**Pull request by nakul1010**
Fixes Part 1 of [#733](https://github.com/Polkadex-Substrate/orderbook/issues/733)

### CODE REVIEW

File: clients/orderbook/rpc/src/lib.rs [hunk 1]
```diff
@@ -63,26 +74,132 @@ pub trait OrderbookApi {
...
+		let trie: TrieDBMut<ExtensionLayout> =
+			TrieDBMutBuilder::from_existing(&mut memory_db, &mut worker_state_root).build();
+
+		let mut ob_recovery_state = ObRecoveryState::new();
+
+		// Generate account info from existing DB
+		for (user_main_account, list_of_proxy_accounts) in all_register_accounts {
+			for asset in allowlisted_asset_ids.clone() {
+				let account_asset = AccountAsset::new(user_main_account.clone(), asset.clone());
+				match trie.get(&account_asset.encode()).unwrap() {
```
Comment (serhii-temchenko) [inline clients/orderbook/rpc/src/lib.rs]: Should be properly handled.

File: clients/orderbook/rpc/src/lib.rs [hunk 2]
```diff
@@ -63,26 +74,132 @@ pub trait OrderbookApi {
...
+					None => {},
+					Some(data) => {
+						let account_balance = Decimal::decode(&mut &data[..]).unwrap();
```
Comment (serhii-temchenko) [inline clients/orderbook/rpc/src/lib.rs]: Should be properly handled.

File: clients/orderbook/rpc/src/lib.rs [hunk 3]
```diff
@@ -63,26 +74,132 @@ pub trait OrderbookApi {
...
+			TrieDBMutBuilder::from_existing(&mut memory_db, &mut worker_state_root).build();
+
+		let mut ob_recovery_state = ObRecoveryState::new();
+
+		// Generate account info from existing DB
+		for (user_main_account, list_of_proxy_accounts) in all_register_accounts {
+			for asset in allowlisted_asset_ids.clone() {
+				let account_asset = AccountAsset::new(user_main_account.clone(), asset.clone());
+				match trie.get(&account_asset.encode()).unwrap() {
+					None => {},
```
Comment (serhii-temchenko) [inline clients/orderbook/rpc/src/lib.rs]: Some log should be here?

File: primitives/orderbook/Cargo.toml [hunk 4]
```diff
@@ -23,6 +23,7 @@ anyhow = { version = "1.0.69", default-features = false }
 rust_decimal = { git = "https://github.com/Polkadex-Substrate/rust-decimal.git", branch = "master", features = ["scale-codec"], default-features = false }
 bls-primitives = {path="../bls-primitives", default-features = false}
 libp2p-core = {version ="0.34.0", optional = true}
+futures = "0.3"
```
Comment (serhii-temchenko) [inline primitives/orderbook/Cargo.toml]: I assume this was added by mistake since not used. Removing it.

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]
I did some minor changes in a latest commits. But please take a look into error handling comments.

**Review by serhii-temchenko — APPROVED** [review]

**Review by Gauthamastro — APPROVED** [review]
General logic looks good, but the code quality needs to be improved, especially inside service.rs and serializing the data in RPC calls. Not a priority.

@nakul1010

**Pull request merged**